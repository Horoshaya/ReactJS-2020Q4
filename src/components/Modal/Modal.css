.wrapper {
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  width: 100%;
  height: 100%;
  justify-content: center;
  align-items: center;
  background: rgba(0, 0, 0, 0.7);
  z-index: 100000;
}

.overlay {
  position: absolute;
  width: 100%;
  height: 100%;
  z-index: 10;
}

.modal {
  position: relative;
  margin: 10px;
  padding: 15px;
  padding-top: 25px;
  width: 300px;
  min-height: 100px;
  background: #232323;
  color: #ffffff;
  z-index: 20;
}

.closeIcon {
  position: absolute;
  right: 0;
  top: 0;
  padding: 15px;
  cursor: pointer;
}

.closeIcon:before {
  content: '\2715';
}
