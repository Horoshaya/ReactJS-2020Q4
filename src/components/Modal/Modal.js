import React from 'react';
import styles from './Modal.css';

const Modal = ({ triggerModal, children }) => {
  return (
    <div className={styles.wrapper}>
      <div className={styles.modal}>
        <span className={styles.closeIcon} onClick={triggerModal}></span>
        {children}
      </div>
      <div className={styles.overlay}></div>
    </div>
  );
};

export default Modal;
