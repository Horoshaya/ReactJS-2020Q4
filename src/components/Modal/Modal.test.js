import React from 'react';
import rerender from 'react-test-renderer';
import Modal from './Modal';

describe('when Modal', () => {
  test('then snapshot created', () => {
    const component = rerender.create(<Modal />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
