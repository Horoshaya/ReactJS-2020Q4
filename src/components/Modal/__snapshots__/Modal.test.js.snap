// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Modal then snapshot created 1`] = `
<div>
  <div>
    <span />
  </div>
  <div />
</div>
`;
